src/cpp/ext/otel/otel_plugin.h

```diff
@@ -65,9 +65,11 @@ class LabelsInjector {
  public:
   virtual ~LabelsInjector() {}
   // Read the incoming initial metadata to get the set of labels to be added to
-  // metrics.
+  // metrics. If \a received_peer_metadata is not nullptr, it is set to true if
+  // peer's metadata was received on the wire, false otherwise.
   virtual std::unique_ptr<LabelsIterable> GetLabels(
-      grpc_metadata_batch* incoming_initial_metadata) = 0;
+      grpc_metadata_batch* incoming_initial_metadata,
+      bool* received_peer_metadata) = 0;
```

Comment:
Instead of exposing this bit in the `LabelsInjector` API and handling this in the OTel call tracers, I think we should instead handle this inside of the `LabelsInjector` impl itself.  The reason I say that is that in the future, if we replace metadata exchange with a metadata lookup service, the concept of "received metadata from peer" will no longer be relevant, and we won't want to have this state machinery stick around.

Reply:
hmmm... that would mean making the LabelsInjector API hold state for the RPC

Reply:
Good point.  Is there a way to do this inside of `MeshLabelsIterable`, not in the `LabelsInjector` itself?

Reply:
I could change the API such that you have to pass back the `LabelsIterable` returned from `GetLabels` in the `AddLabels` function

Reply:
That seems fine for the server side.  But it also has to work on the client side, where `AddLabels()` is called before `GetLabels()`.

Reply:
We don't need this ability on the client anyway, so it'd be fine with a nullptr on the client side.
It's just the server that has to check whether the client sent metadata before it attaches metadata itself, whereas the client always attaches metadata.

Reply:
That sounds fine.

Reply:
Done